source/developers/file-transfer-protocol.rst

```diff
@@ -0,0 +1,228 @@
+
+1. Start MinIO with an FTP and/or SFTP port enabled.
+
+   .. code-block:: shell
+      :class: copyable
+
+      minio server http://server{1...4}/disk{1...4}                               \
+      --ftp="address=:8021" --ftp="passive-port-range=30000-40000"                \
+      --sftp="address=:8022" --sftp="ssh-private-key=/home/miniouser/.ssh/id_rsa" \
+      ...
+    
+   See the :mc-cmd:`minio server --ftp` and :mc-cmd:`minio server --sftp` for details on using these flags to start the MinIO service.
+
+   The output of the command should returns a response that includes something like the following:
+
+   .. code-block:: shell
+
+      MinIO FTP Server listening on :8021
+      MinIO SFTP Server listening on :8022
+
+2. Use your preferred ftp client to connect to the MinIO deployment with the user whose policies you want to use to access buckets and objects.
+
+   The specifics of connecting to the MinIO deployment depend on your FTP client.
+   Refer to the documentation for your client.
+
+   To connect over TLS or through SSH, you must use a client that supports the desired protocol.
+
+Examples
+--------
+
+The examples here use the ``ftp`` CLI client on a Linux system.
+
+Connect to an FTP Server
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following example Connect to a server using ``minio`` credentials to list contents in a bucket named ``runner``
+
+.. code-block:: shell
+
+   > ftp localhost -P 8021
+   Connected to localhost.
+   220 Welcome to MinIO FTP Server
+   Name (localhost:user): minioadmin
+   331 User name ok, password required
+   Password:
+   230 Password ok, continue
+   Remote system type is UNIX.
+   Using binary mode to transfer files.
+   ftp> ls runner/
+   229 Entering Extended Passive Mode (|||39155|)
+   150 Opening ASCII mode data connection for file list
+   drwxrwxrwx 1 nobody nobody            0 Jan  1 00:00 chunkdocs/
+   drwxrwxrwx 1 nobody nobody            0 Jan  1 00:00 testdir/
+   ...
+
+Download an Object
+~~~~~~~~~~~~~~~~~~
+
+The example continues an ftp session already in progress.
+First, we list items in a bucket, then we download the contents of the bucket.
+
+.. code-block:: shell
+
+   ftp> ls runner/chunkdocs/metadata
+   229 Entering Extended Passive Mode (|||44269|)
+   150 Opening ASCII mode data connection for file list
+   -rwxrwxrwx 1 nobody nobody           45 Apr  1 06:13 chunkdocs/metadata
+   226 Closing data connection, sent 75 bytes
+   ftp> get
+   (remote-file) runner/chunkdocs/metadata
+   (local-file) test
+   local: test remote: runner/chunkdocs/metadata
+   229 Entering Extended Passive Mode (|||37785|)
+   150 Data transfer starting 45 bytes
+   	45        3.58 KiB/s
+   226 Closing data connection, sent 45 bytes
+   45 bytes received in 00:00 (3.55 KiB/s)
+   ...
+
+Connect to an SFTP Server
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following example connects to an SSH FTP server, lists the contents of a bucket named ``runner``, and downloads an object.
+
+.. code-block:: text
```

Comment:
shell lexing should be fine here too?

Reply:
It's not. It throws a build error, and I can't figure out why.

Reply:
Swapped back to shell, and will troubleshoot if it continues to throw errors.